### PR TITLE
handle remaining args as single expression

### LINKED ISSUE #10: Parse all arguments
Currently when calling `bitwise 3 + 3`, the result for 3 is displayed.
To avoid quoting (and making usage in pipes easier), all arguments should be parsed as an expression.

Thanks, this is already a great tool!

### PR DESCRIPTION
**Pull request by noerw**
fixes #10

This is pretty much the first time I wrote C, so please tell me if I did something stupid.

### CODE REVIEW

**Comment by mellowcandle** [comment]
I see what you did there, it's a nice hack, it works as long argv[i] are continuous in memory.
I'm not sure what the standard says about it.
I think it will be safer just to allocate a buffer and append the tokens to it one by one.
If the buffer is too small, your can use realloc() to increase it.

**Comment by noerw** [comment]
@mellowcandle Now using a buffer method.

File: src/main.c [hunk 1]
```diff
@@ -149,19 +149,27 @@ int main(int argc, char *argv[])
...
+		uint32_t expr_len = argc - optind - 1; // account for ' ' between args
+		for (uint32_t i = optind; i < argc; i++)
+			expr_len += strlen(argv[i]);
+
+		char *expression = malloc((sizeof 'a') * expr_len);
+		uint32_t expr_pos = 0;
 		for (uint32_t i = optind; i < argc; i++) {
-			char* end = strstr(argv[i], "\0");
-			*(end - 1) = ' ';
+			strcpy(&expression[expr_pos], argv[i]);
```
Comment (mellowcandle) [inline src/main.c]: Please use strncpy

File: src/main.c [hunk 2]
```diff
@@ -149,19 +149,27 @@ int main(int argc, char *argv[])
...
 	init_colors();
 
-	if (optind < argc) {
-		// treat all remaining arguments as a single string for parsing
+	if (optind < argc) { // non-interactive mode
+		uint32_t expr_len = argc - optind - 1; // account for ' ' between args
+		for (uint32_t i = optind; i < argc; i++)
+			expr_len += strlen(argv[i]);
+
+		char *expression = malloc((sizeof 'a') * expr_len);
```
Comment (mellowcandle) [inline src/main.c]: Check for NULL here.

File: src/main.c [hunk 3]
```diff
@@ -149,19 +149,27 @@ int main(int argc, char *argv[])
 
 	init_colors();
 
-	if (optind < argc) {
-		// treat all remaining arguments as a single string for parsing
+	if (optind < argc) { // non-interactive mode
+		uint32_t expr_len = argc - optind - 1; // account for ' ' between args
```
Comment (mellowcandle) [inline src/main.c]: please use int for counters, this is the common idiom

File: src/main.c [hunk 4]
```diff
@@ -149,19 +149,27 @@ int main(int argc, char *argv[])
...
+			expr_len += strlen(argv[i]);
+
+		char *expression = malloc((sizeof 'a') * expr_len);
+		uint32_t expr_pos = 0;
 		for (uint32_t i = optind; i < argc; i++) {
-			char* end = strstr(argv[i], "\0");
-			*(end - 1) = ' ';
+			strcpy(&expression[expr_pos], argv[i]);
+			expr_pos += strlen(argv[i]);
+			strcpy(&expression[expr_pos], " ");
```
Comment (mellowcandle) [inline src/main.c]: no need for function call here
expression[expr_pos++] = ' ';

**Comment by noerw** [comment]
Should have adressed your review.
Thanks, I definitely learned a thing or two from your review!

**Pull request merged**